### DISCUSSION TITLE
Update zoe depth any n and replace the any model it uses with the new any s v2 model

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#131] [General]
Hi,  Thanks for all these great updates by the way. :) Zoe any n is by far the best model on 3d depth, would it be possible if u can retrain the zoe any n to use the new any s version 2 model since the any s is free to use and more accurate on backgrounds and edges.

**Comment by donrikk** [comment]
this is a fantastic idea any s v2 is already stellar in quality but wit hthe addition of zoe depths superior foreground depiction and background accuracy i think we could have a decently fast model that does alot of work under the hood!

**Comment by loawizard** [comment]
yep would be great

**Comment by Ryudoadema** [comment]
Agreed. Appreciate all the updates and sounds like an updated zoe would be another worthwhile one if possible.

**Comment by nagadomi** [comment]
`Depth-Anything-V2-Metric-Hypersim-Large` is a metric depth model tuned for indoor scenes. It is an alternative to ZoeD_Any_N in Depth-Anything-V2.
I will try to see if it works.

**Reply by kaelsonofkrypto** [reply]
man you are absolutely the man , thanks nagadomi, you have some serious talent. I continue to be impressed more and more everyday with what u have built.  that sounds like a solid option, can't wait to see how it works when u implement it. 

Just curious does iw3 use tensor cores? i have noticed that once u start using depth resolutions above 700 it drastically gets rid of most artifacts on any s version. but trying to do this on any of the other models at higher depth resolutions gets drastically slow on other models when using high depth resolution even with low vram checked.  i think the key to getting to artifact free is a depth resolution of between 1800 to 2072, but at the moment u could only achieve something like this with a 48gb gpu. do u think tensor cores would make it where its possible to achieve high depth resolutions like 1800 on 24gb gpu vram and if its not already in iw3 do u think it may help?

**Reply by nagadomi** [reply]
Added `ANY_V2_N` and `ANY_V2_K`. These are alternatives to `ZoeD_ANY_N` and `ZoeD_ANY_K`.
However, the background area looks slightly flatter than ZoeDepth. `foreground-scale = -2` may be closer looking to ZoeDepth.

For install, See https://github.com/nagadomi/nunif/blob/master/iw3/README.md#about-any_v2_b-any_v2_l-any_v2_n-any_v2_k , https://huggingface.co/depth-anything/Depth-Anything-V2-Metric-Hypersim-Large/tree/main

>Tensor Core

It depends on the GPU, but in FP16 mode it should use Tensor Core.
https://discuss.pytorch.org/t/does-pytorch-use-tensor-cores-by-default/167676/2

**Reply by loawizard** [reply]
Depth-Anything-V2-Metric-Hypersim-Large is AMAZING in IW3!  It's slow though , i asked the devs of Depth anything if we can have a small version of it

**Reply by loawizard** [reply]
Thank you for letting us implement all of these amazing depthmaps!

**Reply by kaelsonofkrypto** [reply]
> `Depth-Anything-V2-Metric-Hypersim-Large` is a metric depth model tuned for indoor scenes. It is an alternative to ZoeD_Any_N in Depth-Anything-V2. I will try to see if it works.


BROOOOOOO, this is the first time i made a perfect 3d conversion with no artifacts, but to be fair 1.5 depth was what i used which is lower,  the new anyv2 N model is the best so far, it could be i got lucky on the first try with it, but from what i can tell its even better than any s v2, and any large v2. i just did a conversion and on fp16 it was perfect and i never do fp16, this was the first time i ever did it on a whole conversion. I HAD ZERO ARTIFACTS at a depth of 724 with 1.5 depth, i have used other models like any l v2 and still had artifacts even with high depth resolutions. this is the best model officially. iw3 is almost perfect, if u can add an auto save feature that allows you to resume if you pause but also auto saves your progress like every 200 frames so in the event a power outage happens u dont lose your conversion process and u can resume soon as u open it backup, this way you don't have to start over from scratch if u could add those two things a pause and an auto save every 200 frames function that u can resume in the event the program crashed or power outage where u could resume, that would honestly bring it closer to a perfect program, other than that, im  hoping for faster speed on tta. Thank you so much for everything really.

**Reply by Ryudoadema** [reply]
> > `Depth-Anything-V2-Metric-Hypersim-Large` is a metric depth model tuned for indoor scenes. It is an alternative to ZoeD_Any_N in Depth-Anything-V2. I will try to see if it works.
> 
> BROOOOOOO YOU HAVE NOW ACHIEVED PERFECT 3D, this is a first, the new anyv2 N model is the best, its even better than any s v2, and any large v2. i just did a conversion and on fp16 it was perfect and i never do fp16, this was the first time i ever did it on a whole conversion. I HAD ZERO ARTIFACTS at a depth of 728, i have used other models like any l and still had artifacts even with high depth resolutions. you found the perfect model. Now literally the only thing i think you really have left to make iw3 completely perfect is an auto resume feature that saves your progress like every 200 frames so in the event a power outage happens u dont lose your conversion process and have to start over from scratch if u could add that, that would honestly be my last request because every thing else is now perfect besides that one thing.

Yes, but how is the jiggle wiggle wobble bobble?

**Reply by kaelsonofkrypto** [reply]
this model was the first time i didnt have the jiggle but definitely i need to do more tests with other types of videos to see if i continue to have luck with no artifacts on the any depth v2 N model

**Comment by kaelsonofkrypto** [comment]
update 2: i had an infamous clip that always struggled with every model i tried on it, it would have artifacts, this clip was so bad that my friends called it the devil clip.  well guess what , today, a historic monumental event has happened for me lol,  the any v2 N model was the first model that rendered it almost perfect on the four second clip, Just WOW, thank u for this model nagadomi. my friends also did some other tests, we did notice at 1.8 depth nose clipping happens, also some other friends tested their clips and are getting wobbles in some situations so i guess there is still some artifacts but its definitely improved over version 1. so at the moment it looks like 1.5 depth is the closest to artifact free. so i guess the next stage is also bringing iw3 to a level where u can get close to artifact free at high depths above 1.5 and hopefully even has 3.0 and 4.0
